### PR TITLE
barbican: restore missing keystone config (bsc#1024554)

### PR DESCRIPTION
**Pull request by Itxaka**
Seems that it was wrongly removed during the minify of the templates.

### CODE REVIEW

File: chef/cookbooks/barbican/templates/default/barbican.conf.erb [hunk 1]
```diff
@@ -24,4 +24,18 @@ control_exchange = 'keystone'
 
 [simple_crypto_plugin]
 
-kek = <%= @kek %>
\ No newline at end of file
+kek = <%= @kek %>
+
+[keystone_authtoken]
+
```
Comment (dirkmueller) [inline chef/cookbooks/barbican/templates/default/barbican.conf.erb]: extra empty line, remove

File: chef/cookbooks/barbican/templates/default/barbican.conf.erb [hunk 2]
```diff
@@ -24,4 +24,18 @@ control_exchange = 'keystone'
...
+signing_dir = /var/cache/barbican/keystone-signing
```
Comment (dirkmueller) [inline chef/cookbooks/barbican/templates/default/barbican.conf.erb]: not needed anymore (only for PKI)

File: chef/cookbooks/barbican/templates/default/barbican.conf.erb [hunk 3]
```diff
@@ -24,4 +24,18 @@ control_exchange = 'keystone'
...
+auth_type = password
+user_domain_id = default
```
Comment (dirkmueller) [inline chef/cookbooks/barbican/templates/default/barbican.conf.erb]: user_domain_name = <%= @keystone_settings["admin_domain"] %>

File: chef/cookbooks/barbican/templates/default/barbican.conf.erb [hunk 4]
```diff
@@ -24,4 +24,18 @@ control_exchange = 'keystone'
...
+project_domain_id = default
```
Comment (dirkmueller) [inline chef/cookbooks/barbican/templates/default/barbican.conf.erb]: project_domain_name = <%= @keystone_settings["admin_domain"]%>

File: chef/cookbooks/barbican/templates/default/barbican.conf.erb [hunk 5]
```diff
@@ -24,4 +24,18 @@ control_exchange = 'keystone'
...
+auth_url = <%= @keystone_settings["admin_auth_url"] %>
```
Comment (dirkmueller) [inline chef/cookbooks/barbican/templates/default/barbican.conf.erb]: auth_url = <%= @keystone_settings['internal_auth_url'] %

File: chef/cookbooks/barbican/templates/default/barbican.conf.erb (hunk 4 above)
Comment (vuntz) [inline chef/cookbooks/barbican/templates/default/barbican.conf.erb]:
```
project_domain_name = <%= @keystone_settings["admin_domain"]%>
user_domain_name = <%= @keystone_settings["admin_domain"] %>
```

File: chef/cookbooks/barbican/templates/default/barbican.conf.erb (hunk 2 above)
Comment (vuntz) [inline chef/cookbooks/barbican/templates/default/barbican.conf.erb]: Isn't that in the conf file created by the package?

File: chef/cookbooks/barbican/templates/default/barbican.conf.erb (hunk 5 above)
Comment (vuntz) [inline chef/cookbooks/barbican/templates/default/barbican.conf.erb]: s/admin/internal/

**Comment by dirkmueller** [comment]
please also include the bugzilla reference (bsc#1024554 I guess?)

File: chef/cookbooks/barbican/templates/default/barbican.conf.erb [hunk 6]
```diff
@@ -24,4 +24,18 @@ control_exchange = 'keystone'
...
+auth_uri = <%= @keystone_settings["public_auth_url"] %>
+auth_version = <%= @keystone_settings["api_version_for_middleware"] %>
+username = <%= @keystone_settings["service_user"] %>
+password = <%= @keystone_settings["service_password"] %>
+project_name = <%= @keystone_settings["service_tenant"] %>
+insecure = <%= @keystone_settings["insecure"] %>
```
Comment (vuntz) [inline chef/cookbooks/barbican/templates/default/barbican.conf.erb]: Missing:
```
region_name = <%= @keystone_settings['endpoint_region'] %>
```

**Comment by Itxaka** [comment]
Looks like they added the keystone section to the conf after newton was cut :o
https://github.com/openstack/barbican/commit/9a3d1c6f7a8e9fdbbd3f61dd9bdc054dd3421c29

**Pull request merged**